File: app/src/main/java/io/temco/guhada/data/model/SnsUser.java
package io.temco.guhada.data.model;

import com.google.gson.annotations.SerializedName;

public class SnsUser extends User{
    @SerializedName("profileJson")
    private UserProfile userProfile;

    private String snsId;
    private String type;

    public SnsUser() {
    }

    public UserProfile getUserProfile() {
        return userProfile;
    }

    public void setUserProfile(UserProfile userProfile) {
        this.userProfile = userProfile;
    }


    public String getSnsId() {
        return snsId;
    }

    public void setSnsId(String snsId) {
        this.snsId = snsId;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}

File: app/src/main/java/io/temco/guhada/data/model/user/SnsUser.kt
package io.temco.guhada.data.model.user

import com.google.gson.annotations.SerializedName
import io.temco.guhada.data.model.user.User
import io.temco.guhada.data.model.user.UserProfile
import java.io.Serializable

/**
 * SNS 회원 정보 클래스
 * @see UserProfile
 * @author Hyeyeon Park
 */
class SnsUser : User(), Serializable {
    @SerializedName("profileJson")
    var userProfile: UserProfile? = UserProfile()

    var snsId: String? = ""
    var snsType: String? = ""
    val profileJsonString = "string"
}